tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/IbmMq/PutIntegration.cs

```diff
@@ -0,0 +1,72 @@
+// <copyright file="PutIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.IbmMq
+{
+    /// <summary>
+    /// IBM MQ Put instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = IbmMqConstants.IbmMqAssemblyName,
+        TypeName = IbmMqConstants.MqDestinationTypeName,
+        MethodName = "Put",
+        ReturnTypeName = ClrNames.Void,
+        ParameterTypeNames = new[] { IbmMqConstants.MqMessageTypeName, IbmMqConstants.MqMessagePutOptionsTypeName },
+        MinimumVersion = "9.0.0",
+        MaximumVersion = "9.*.*",
+        IntegrationName = IbmMqConstants.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class PutIntegration
+    {
+        internal static CallTargetState OnMethodBegin<TTarget, TMessage, TOptions>(TTarget instance, TMessage msg, TOptions options)
+        where TMessage : IMqMessage
+        where TTarget : IMqQueue
+        {
+            if (((IDuckType)instance).Instance == null || ((IDuckType)msg).Instance == null)
+            {
+                return CallTargetState.GetDefault();
+            }
```

Comment:
You can simplify these by adding `IDuckType` to the constraints instead of casting:

```suggestion
        where TMessage : IMqMessage, IDuckType
        where TTarget : IMqQueue, IDuckType
        {
            if (instance.Instance == null || msg.Instance == null)
            {
                return CallTargetState.GetDefault();
            }
```

Alternatively, you can make the interfaces themselves directly implement IDuckType:

```csharp
interface IMqMessage :  IDuckType
{
  // ...
}
```